src/graph_notebook/magics/graph_magic.py

```diff
@@ -163,37 +165,54 @@ def sparql(self, line='', cell='', local_ns: dict = None):
         args = parser.parse_args(line.split())
         mode = str_to_query_mode(args.query_mode)
         tab = widgets.Tab()
+        titles = []
+        children = []
+
+        metadata = {
+            "Query mode": args.query_mode,
```

Comment:
Can we consider making a metadata class we instantiate, and have something that builds it into the output we want? It should help us minimize the amount of code we have to add to each magic command for us to have this type of data.